src/main/java/com/google/devtools/build/lib/analysis/ToolchainResolver.java

```diff
@@ -521,17 +522,22 @@ public ToolchainContext load(Iterable<ConfiguredTargetAndData> toolchainTargets)
       ImmutableList.Builder<TemplateVariableInfo> templateVariableProviders =
           new ImmutableList.Builder<>();
       for (ConfiguredTargetAndData target : toolchainTargets) {
-        Label discoveredLabel = target.getTarget().getLabel();
+        Label discoveredLabel;
+        // Aliases are in toolchainTypeToResolved by the original alias label, not via the final
+        // target's label.
+        if (target.getConfiguredTarget() instanceof AliasConfiguredTarget) {
```

Comment:
Does this support chains of `alias` rules? I'm wondering if this part should be recursive.

Reply:
It only needs the top of the chain, since that's the label used in the map (the previous code actually traversed the chain, which was the problem). I've added a test to prove this.